runtime/j9vm/jvm.c

```diff
@@ -1496,24 +1506,131 @@ static jint initializeReflectionGlobals(JNIEnv * env, BOOLEAN includeAccessors)
 	return JNI_OK;
 }
 
-
-
 /**
- *  jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args)
- *  Load and initialize a virtual machine instance.
- *	This provides an invocation API that runs the J9 VM in BFU/sidecar mode
+ * jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args)
+ * Load and initialize a virtual machine instance.
+ * This provides an invocation API that runs the J9 VM in BFU/sidecar mode.
  *
- *  @param pvm pointer to the location where the JavaVM interface
+ * @param pvm pointer to the location where the JavaVM interface
  *			pointer will be placed
- *  @param penv pointer to the location where the JNIEnv interface
+ * @param penv pointer to the location where the JNIEnv interface
  *			pointer for the main thread will be placed
- *	@param vm_args java virtual machine initialization arguments
+ * @param vm_args java virtual machine initialization arguments
  *
- *  @returns zero on success; otherwise, return a negative number
+ * @returns zero on success; otherwise, return a negative number
  *
- *	DLL: jvm
+ * DLL: jvm
  */
 jint JNICALL JNI_CreateJavaVM(JavaVM **pvm, void **penv, void *vm_args) {
+	return JNI_CreateJavaVM_impl(pvm, penv, vm_args, FALSE);
+}
+
+#if defined(JITSERVER_SUPPORT)
+int32_t JNICALL
+JITServer_CreateServer(JITServer **jitServer, void *serverArgs)
+{
+	JNIEnv *env = NULL;
+	JITServer *server = NULL;
+	jint rc = JITSERVER_OK;
+
+	server = malloc(sizeof(JITServer));
+	if (NULL == server) {
+		rc = JITSERVER_OOM;
+		goto _end;
+	}
+	server->startJITServer = startJITServer;
+	server->waitForJITServerTermination = waitForJITServerTermination;
+	rc = JNI_CreateJavaVM_impl(&server->jvm, (void **)&env, serverArgs, TRUE);
+
+	if (JNI_OK != rc) {
+		rc = JITSERVER_CREATE_FAILED;
+		goto _end;
+	}
+	*jitServer = server;
+	rc = JITSERVER_OK; /* success */
+
+_end:
+	if ((JITSERVER_OK != rc) && (NULL != server)) {
+		free(server);
```

Comment:
When does the `server = malloc(sizeof(JITServer));` get freed in the success case?  I don't see it in the `waitForJITServerTermination` call.  Is there is a missing `JITServer_DestroyServer` call?

Reply:
Also, I don't see where the JavaVM is destroyed...

Reply:
`JITServer_DestroyServer` is not there, as there is no way to call it for now. The main thread just keeps waiting for the JITServer listener thread which does not exit. To terminate JITServer, we need to kill the process.
This is not ideal, but it has been like this. As per my discussion with @mpirvu  we may add support to gracefully terminate the server in future.

Reply:
Please open an issue for adding the `JITServer_DestroyServer` api.  This PR can be merged once the issue is opened for this.